app/v1/controllers/ListBenefitsController.scala

```diff
@@ -103,4 +103,19 @@ class ListBenefitsController @Inject()(val authService: EnrolmentsAuthService,
       case DownstreamError => InternalServerError(Json.toJson(errorWrapper))
     }
   }
+
+  private def process(response: ListBenefitsResponse[StateBenefit], benefitId: Option[String]): ListBenefitsResponse[StateBenefit] = {
+
+    benefitId match {
+      case None => response
+      case _ => (response.stateBenefits, response.customerAddedStateBenefits) match {
+        case (Some(hmrc), Some(custom)) if hasCommon(hmrc, custom) != Nil =>
+          response.copy(Some(hmrc.map(_.copy(isCommon = true))), Some(custom.map(_.copy(isCommon = true))))
+        case (_, _) => response
+      }
+    }
+  }
+
+  private def hasCommon(hmrcB: Seq[StateBenefit], customB: Seq[StateBenefit]): List[StateBenefit] =
```

Comment:
also this function doesnt return a boolean, which you would expect it to by its name.  If what we are doing here is retrieving common benefits between the two lets call it something like 'listCommonBenefits'